### ISSUE TITLE
ENH: Allow rank to return int64 for numpy types when no missing values are present

### ISSUE THREAD

**Issue by mroeschke** [pandas-dev/pandas#55598]
```python
In [2]: import pandas as pd

In [3]: df = pd.DataFrame({'a': [1, 1, 1, 2, 2], 'b': [1,2 ,3, 4, 5]})

In [4]: df.groupby('a').rank(method='min')
Out[4]: 
     b
0  1.0
1  2.0
2  3.0
3  1.0
4  2.0

In [5]: df.rank()
Out[5]: 
     a    b
0  2.0  1.0
1  2.0  2.0
2  2.0  3.0
3  4.5  4.0
4  4.5  5.0
```

In these cases, it would be more natural to return int64 instead of float64

**Comment by rhshadrach** [comment]
Agreed in all cases except for `method='average'`. Here, the value may need to be float in the case of ties (e.g. `df['a']` in the example above). If we were to return integers in the case of no ties, this would be value-dependent behavior. I think we should therefore prefer to always return float.

**Comment by Cappuchinoo** [comment]
@rhshadrach Just making sure: when you said we should always prefer to return float were you talking exclusively about the specific case of `method='average'` or about any call of the rank function?

**Comment by rhshadrach** [comment]
Only the case of `method='average'`.

**Comment by Cappuchinoo** [comment]
Cool, I'll give this a shot. Appreciate the quick response!

**Comment by Cappuchinoo** [comment]
take

**Comment by Cappuchinoo** [comment]
Hey, I made what I think is what was expected, would you mind taking a look just to let me know if I'm in the right path so I can change the tests to reflect the changes? Thanks!
